index.js

```diff
@@ -4,6 +4,11 @@ module.exports = {
   },
 
   responseObject: function(response) {
+    try {
+      JSON.parse(response.xhr.responseText);
+    } catch(err) {
+      return {invalid_JSON: response.xhr.responseText};
```

Comment:
The consumer can handle a syntax error by matching on the error string in the consumers failure state.

```javascript
getSomeXhr()
  .then(undefined, function(err) {
    if (err.message.contains('syntax error')) {
      // handle error
    }
    // The error is not a syntax error, so we need to re-throw it.
    throw err;
  });